src/OrbitBase/TypedefTest.cpp

```diff
@@ -324,4 +325,34 @@ TEST(Typedef, ComparisonIsCorrect) {
   EXPECT_GT(MyType<int>(kGreater), MyType<int>(kLesser));
 }
 
+struct WrapperWithPlusTag : PlusTag {};
+
+template <typename T>
+using WrapperWithPlus = Typedef<WrapperWithPlusTag, T>;
+
+TEST(Typedef, WrapperWithPlusHasPlus) {
+  constexpr int kAValue = 1;
+  constexpr int kBValue = 2;
+  WrapperWithPlus<int> a(kAValue);
+  WrapperWithPlus<int> b(kBValue);
+  EXPECT_EQ(*(a + b), kAValue + kBValue);
+}
+
+TEST(Typedef, WrapperWithPlusHasPlusAndPromotes) {
+  constexpr int kInt = 1;
+  constexpr float kFloat = 0.5;
+  WrapperWithPlus<int> a(kInt);
+  WrapperWithPlus<float> b(kFloat);
+  WrapperWithPlus<float> result = a + b;
+  EXPECT_EQ(*result, kInt + kFloat);
+}
+
+TEST(Typedef, WrapperWithPlusHasPlusAndConvertsArgument) {
+  constexpr std::chrono::nanoseconds kNanos(1000);
+  constexpr std::chrono::microseconds kMicros(1);
+  WrapperWithPlus<std::chrono::nanoseconds> a(kNanos);
+  WrapperWithPlus<std::chrono::microseconds> b(kMicros);
+  EXPECT_EQ(*(a + b), kNanos + kMicros);
+}
+
```

Comment:
A test for a move-only type would be great. This helps to test perfect forwarding even though it's not that likely to happen.